### PR TITLE
Using time crate instead of chrono

### PR DESCRIPTION
**Pull request by HallerPatrick**
This PR solves Issue #526.

I ported, more or less, the `chrono-humanize` library to only use the std lib time and as a optional feature the `time-rs` crate ( for display iso time). Because I wanted to keep most of tests of the original repo (which are a lot!) and also included some nice interface, I made this an [own crate](https://github.com/HallerPatrick/time-humanize).

The code now even looks way cleaner and include less logic. I tested the human readable output as well as the iso output on different repos. Feel free to test this  as well.

### CODE REVIEW

**Comment by HallerPatrick** [comment]
Looks like the audit finds a soundness issue for `chrono` and `time`. `time` is an optional feature of `time-humanize`, that is used to display the time as the iso standard. I am not sure if this is affecting our use case.

File: Cargo.toml [hunk 1]
```diff
@@ -30,8 +30,8 @@ anyhow = "1.0"
 askalono = "0.4.3"
 byte-unit = "4.0.12"
 bytecount = "0.6.2"
-chrono = "0.4"
-chrono-humanize = "0.2.1"
+time-humanize = {version="0.1.2", features = ["time"]}
+time = { version = "0.3.5", features = ["formatting"]}
```
Comment (spenserblack) [inline Cargo.toml:34]:
```suggestion
time-humanize = { version="0.1.2", features = ["time"] }
time = { version = "0.3.5", features = ["formatting"] }
```
Minor nitpick. I don't think we have a preferred style, but I've usually seen spaces inside braces.
For example, [the docs for dependencies in `Cargo.toml`](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-other-registries).

File: src/info/repo.rs [hunk 2]
```diff
@@ -337,23 +339,21 @@ fn bytes_to_human_readable(bytes: u128) -> String {
...
-            .to_rfc3339_opts(chrono::SecondsFormat::Secs, true)
+        to_rfc3339(HumanTime::from(time.seconds()))
     } else {
-        let ht = HumanTime::from(dt_with_tz);
+        let ht = HumanTime::from_duration_since_timestamp(time.seconds().unsigned_abs());
         format!("{}", ht)
     }
 }
 
+fn to_rfc3339<T>(dt: T) -> String
```
Comment (spenserblack) [inline src/info/repo.rs:350]: It might be cool if we added some tests for these functions. Unfortunately we didn't have any prior tests for duration formatting to confirm consistent behavior.

**Review by spenserblack — COMMENTED** [review]
Looks like `chrono` is still in our dependency tree because it's a build dependency of `tokei`, but I think that's OK.
I didn't see the advisory for `time` that you mentioned, though.

File: Cargo.toml (hunk 1 above)
Comment (spenserblack) [reply Cargo.toml]: Actually ignore the above nitpick, since my suggestion doesn't match the rest of `Cargo.toml` :sweat_smile: 
If anything, it would be more like
```suggestion
time-humanize = {version="0.1.2", features = ["time"]}
time = {version = "0.3.5", features = ["formatting"]}
```

File: src/info/repo.rs (hunk 2 above)
Comment (HallerPatrick) [reply src/info/repo.rs:350]: Yes, can do that

**Comment by spenserblack** [comment]
Sorry @HallerPatrick, I gave you a bad suggestion. See https://github.com/o2sh/onefetch/pull/533#discussion_r751300809

**Comment by HallerPatrick** [comment]
> Looks like `chrono` is still in our dependency tree because it's a build dependency of `tokei`, but I think that's OK.
> I didn't see the advisory for `time` that you mentioned, though.

In the audit there is a reference link, which leads to the `time-rs` repo. `chrono` and `time` have the same unsoundness bug. [time-rs issue](https://github.com/time-rs/time/issues/293) , [chrono issue](https://github.com/chronotope/chrono/issues/499)

**Comment by spenserblack** [comment]
Got it, thanks for linking the issue!
According to [this security advisory](https://github.com/time-rs/time/security/advisories/GHSA-wcg3-cvx6-7396) and https://github.com/time-rs/time/issues/293#issuecomment-731504605 it seems that `time >= 0.2.23` won't be affected. In this PR both `onefetch` and `time-humanize` depend on `0.3.5`, so it looks OK in my opinion.
The affected `time` version (`0.1`) comes from chrono, which through the chain of dependencies is a build dependency of tokei. I think that it's out of our scope to fix deeply nested build dependencies of one of our dependencies :slightly_smiling_face: 

tl;dr it doesn't seem like we have to worry about `time`'s security advisory IMO

File: src/info/repo.rs (hunk 2 above)
Comment (HallerPatrick) [reply src/info/repo.rs:350]: @spenserblack You think those tests are sufficient?

Comment (spenserblack) [reply src/info/repo.rs:350]: Yeah, those look good to me :+1:
Thanks!

**Review by spenserblack — APPROVED** [review]
Thanks for adding the tests, LGTM!

**Comment by HallerPatrick** [comment]
@o2sh does not seem to be active currently. Because I got your okay, I would merge this now

**Comment by o2sh** [comment]
So sorry I couldn't find the time to review your PR @HallerPatrick :disappointed: 
Thanks @spenserblack for providing great feedback as always :muscle: 

Everything seems to work as expected. 

My only concern is the `edition = "2021"` used by [time-humanize](https://github.com/HallerPatrick/time-humanize/blob/ef3668f61cdbd327f6a9d360b786ab36c865d1e4/Cargo.toml#L8) which makes onefetch's msrv (minimum supported rust version) go from 1.50.0 to 1.56.1

Rust version 1.56.1 is [very recent](https://blog.rust-lang.org/2021/11/01/Rust-1.56.1.html) and not yet available for [launchpad](https://launchpad.net/ubuntu/+source/rustc)

**Comment by HallerPatrick** [comment]
> So sorry I couldn't find the time to review your PR @HallerPatrick :disappointed: 
> 
> Thanks @spenserblack for providing great feedback as always :muscle: 
> 
> 
> 
> Everything seems to work as expected. 
> 
> 
> 
> My only concern is the `edition = "2021"` used by [time-humanize](https://github.com/HallerPatrick/time-humanize/blob/ef3668f61cdbd327f6a9d360b786ab36c865d1e4/Cargo.toml#L8) which makes onefetch's msrv (minimum supported rust version) go from 1.50.0 to 1.56.1
> 
> 
> 
> Rust version 1.56.1 is [very recent](https://blog.rust-lang.org/2021/11/01/Rust-1.56.1.html) and not yet available for [launchpad](https://launchpad.net/ubuntu/+source/rustc)

No problem at all. @spenserblack gave good feedback.

For `time-humanize`: I was not aware of the Edition difference. I will look to lower it to `onefetch`s version if possible.

**Comment by HallerPatrick** [comment]
@o2sh I updated the rust version of `time-humanize` to `edition = "2018"`. Would you update the dependency to `time-humanize = { version="0.1.3", features = ["time"] }`? Or should I do this through a PR?

**Comment by o2sh** [comment]
Thanks @HallerPatrick 👍 
I think we can wait for the dependabot to take care of it 🤖
